### ISSUE TITLE
Adding bridge to a project breaks "nuget pack" (unable to reading metadata)

### ISSUE THREAD

**Issue by DanTup** [bridgedotnet/Bridge#1894]
To repro:

1. Create a new solution with two class libraries (`bridgetest` and `nonbridgetest`)
2. Set the assembly version for both projects to `1.2.3.4`
3. Add Bridge to `bridgetest` via nuget
4. From the command line in the solution folder run `nuget pack nonbridgetest\nonbridgetest.csproj`
5. From the command line in the solution folder run `nuget pack bridgetest\bridgetest.csproj`

For the Bridge project, you'll see this in the output:

> WARNING: Unable to extract metadata from 'BridgeTest.dll'.

and you will end up with two different versioned nupkgs:

- bridgetest.1.0.0.0.nupkg
- nonbridgetest.1.2.3.4.nupkg

It seems like something in the DLL that's written by Bridge is causing nuget to fail to read the metadata. The correct version number does still show up in the Windows file properties dialog, so the metadata is there, but presumably some difference is causing nuget to fail to read.

Unfortunately nuget swallows the original exception so I'm not sure what's happening, though I expect it's failing inside `GetMetadata` [from here](https://github.com/NuGet/NuGet2/blob/master/src/Core/Authoring/AssemblyMetadataExtractor.cs)

**Comment by DanTup** [comment]
@geoffreymcgill @LeonidVeriga I don't support there was any investigation into this? I'm trying to set up more packages on our NuGet feed today and we're hitting the same problem still:

```text
2017-03-06 12:14:51,967 [99:DEBUG] [SecretProject nuget] Packing files from 'D:\CI\W\SecretProject\SecretProject\bin\Release'.
2017-03-06 12:14:51,998 [99:DEBUG] [SecretProject nuget] WARNING: Unable to extract metadata from 'SecretProject.dll'.
```

**Comment by LeonidVeriga** [comment]
@DanTup, I think this is caused by these project settings:
```xml
<PropertyGroup>
  <NoStdLib>True</NoStdLib>
  <AddAdditionalExplicitAssemblyReferences>false</AddAdditionalExplicitAssemblyReferences>
  <AdditionalExplicitAssemblyReferences />
</PropertyGroup>
```
May I suggest a workaround to use packaging with **nuspec** file instead of **csproj** file, like:
```
nuget pack "test.nuspec" -Version "1.2.3.4" -Properties "Configuration=Dubug;Platform=AnyCPU"
```

If this cmd is run inside of msbuild process then the **-Version** parameter can be grabbed automatically.

**Comment by DanTup** [comment]
I was trying hard to avoid a nuspec file to avoid duplicating a bunch of info ;( It's hard for us to run as part of msbuild without changes (it normally runs as a step from our build server).

Based on the stuff you posted, I presume this isn't likely going to be an easy fix? If not, I guess it makes sense to switch to a nuspec!

**Comment by LeonidVeriga** [comment]
Yes, I think there is no workaround to implement inside Bridge. Some utils can load and read **NoStdLib** assembly metadata (like we use Mono.Cecil inside Bridge or JetBrains dotPeek).

There is an issue reported by @Dia6lo in the NuGet repo - https://github.com/NuGet/Home/issues/2360
The current milestone is Future-1 that does not look as a priority.

**Comment by geoffreymcgill** [comment]
@DanTup – This is something I'm investigating. We need to make some changes to the compilation process, and we're hoping those changes will fix this issue.

No time frame and no guarantees, but we do our best to try and get this working.
